test/direct_mlj_interface.jl

```diff
@@ -0,0 +1,77 @@
+using Random: Random
+import Random.seed!
+using MLJBase: MLJBase, categorical
+using Flux
+using StableRNGs
+using MLJ
+using MLJ:predict,fit!
+using LaplaceRedux
+
+cv = CV(; nfolds=3)
+
+@testset "Regression" begin
+    flux_model = Chain(
+        Dense(4, 10, relu),
+        Dense(10, 10, relu),
+        Dense(10, 1)
+    )
+    model = LaplaceRegressor(model=flux_model,epochs=50)
+    
+    X, y = make_regression(100, 4; noise=0.5, sparse=0.2, outliers=0.1)
+    mach = machine(model, X, y) #|> MLJBase.fit! #|> (fitresult,cache,report)
+    MLJBase.fit!(mach,verbosity=1)
+    Xnew, _ = make_regression(3, 4; rng=123)
+    yhat = MLJBase.predict(mach, Xnew) # probabilistic predictions
+    MLJBase.predict_mode(mach, Xnew)   # point predictions
+    MLJBase.fitted_params(mach)   #fitted params function 
+    MLJBase.training_losses(mach) #training loss history
+    model.epochs= 100 #changing number of epochs
+    MLJBase.fit!(mach) #testing update function
+    model.epochs= 50 #changing number of epochs to a lower number
+    MLJBase.fit!(mach) #testing update function
+    model.fit_prior_nsteps = 200 #changing LaplaceRedux fit steps
+    MLJBase.fit!(mach) #testing update function (the laplace part)
+    # evaluate!(mach, resampling=cv, measure=l2, verbosity=0)
```

Comment:
This is the only thing that's still failing @pasq-cat because a measure like `l2` cannot be applied to the probabilistic predictions of the `LaplaceRegressor`. Perhaps `MMI.predict` should return the mode? But then I'm not sure how we'd ever access the predicted distributions in the MLJ context. This might be tricky ...

Reply:
@pat-alt  uhm there are other loss functions already implemented , i guess one of them should be good https://juliaai.github.io/MLJ.jl/stable/performance_measures/

otherwise we overload evaluate! and set a custom measure (it should be the heteroscedastic negative log likelihood, no?)

Reply:
@pat-alt  also the documentation check will always fail because to follow MLJ rules i had to remove the docstrings on the structs because they conflicted with the long strings at the bottom